src/helics/application_api/Federate.cpp

```diff
@@ -1155,4 +1155,12 @@ void Federate::setFilterOption (const Filter &filt, int32_t option, bool option_
     coreObject->setHandleOption (filt.getHandle (), option, option_value);
 }
 
+    void Federate::setInfo(interface_handle handle, const std::string& info) {
+        coreObject->setInterfaceInfo(handle, info);
+    }
+
+std::string const &Federate::getInfo(interface_handle handle) {
+    return coreObject->getInterfaceInfo(handle);
+}
+
```

Comment:
I am wondering whether this is best or if a series of overloads might be better.  Using the interface_handle directly  does simplify the code so I am not sure but it might be inconsistent with the rest of the API.  Any thoughts?

Reply:
Would it be more consistent to have a template method in the Federate that takes one of the object types and calls getHandle() internally then passes it to the core? That occurred to me as an option last night.

Reply:
I have thought about it too, but what I am trying to avoid is having to include the definitions of Inputs, Publications, Endpoints, and Filters in the header file that defines the Federate.    There may be a way to have the template definition in a separate file but that gets a little complicated.

Reply:
There are a couple other functions that are identical between all the interfaces.  I thought about making all the interfaces share a base class but that has similar issues to the templates